### ISSUE TITLE
Some concerns about the rdp_accountant.py

### ISSUE THREAD

**Issue by AshiakerWang** [tensorflow/privacy#104]
Thank you for open-sourcing the code for calculating the privacy guarantee achieved. However, while going through the` rdp_accountant.py`, I got some questions.

It can be found that the function `compute_rdp(q, noise_multiplier, steps, orders)`  calls the function `_compute_rdp(q, sigma, alpha)`  to calculate the RDP guarantee, and the input parameters of the two functions are `(q, sigma, alpha)`  and` (q, noise_multiplier, steps, orders)` respectively, which indicates that you have treated the parameter `sigma`  the same as` noise_multiplier`.

**However, I hold the opinion that  ` noise_multiplier`  and  `sigma` are different and cannot be used 
 interchangeably.**

According to the comments in `rdp_accountant.py`:

- ` noise_multiplier`  is "the ratio of the standard deviation of the Gaussian noise to the l2-sensitivity of the function to which it is added".

- while `sigma`  is "the standard deviation of the additive Gaussian noise".

It can be found that :

- ` noise_multiplier` =`sigma`/`l2-sensitivity of the function`

To end with, I am afraid  `l2-sensitivity of the function` is a quantity related to the `l2 clip bound` according to [Dwork's book](https://www.cis.upenn.edu/~aaroth/Papers/privacybook.pdf), Which is not always 1, thus ` noise_multiplier` !=`sigma`.

I hope I have clarified my concerns and it will be appreciated if you could resolve my confusion.

**Comment by kairouzp** [comment]
You are right that in general, `noise_multiplier` and `sigma` are not the same. We should have used `noise_multiplier` instead of `sigma` here for the private function `_compute_rdp(q, sigma, alpha)`. It should instead be `_compute_rdp(q, noise_multiplier, alpha)`. However, rest assured that there is no mistake here because `compute_rdp(q, noise_multiplier, steps, orders)` is passing the `noise_multiplier` to `_compute_rdp(q, sigma, alpha)`. Hope this helps.

**Comment by AshiakerWang** [comment]
Thank you very much for your response and explanation.

However, I am really sorry for didn't state my question clearly.

In one word, I hold the opinion that `Line 259` and `Line 261`(the snippets below) in `rdp_accountant.py` should be replaced with:

-  `   rdp = _compute_rdp(q, noise_multiplier * L2_sensitivity, orders) ` and 

- `rdp = np.array([_compute_rdp(q, noise_multiplier * L2_sensitivity, order) for order in orders]) `

respectively, since  `noise_multiplier =sigma / l2-sensitivity of the function`(defined in the code).

https://github.com/tensorflow/privacy/blob/da9fb28b37535a66bb4092843c965f6e21667e2e/tensorflow_privacy/privacy/analysis/rdp_accountant.py#L258-L262

In other words, `compute_rdp(q, noise_multiplier, steps, orders)` **should not pass the `noise_multiplier` directly** to `_compute_rdp(q, sigma, alpha)`  without considering the `L2_sensitivity` of the function.

Hope  this could clarify my concerns.

**Comment by kairouzp** [comment]
Please see proposition 7 of this paper: https://arxiv.org/pdf/1702.07476.pdf. 

The alfa-RDP of two Gaussian (with std = sigma) that are shifted apart by  l2_sensitivity apart is: alfa/2*(1/noise_multiplier^2). 

Does this address your concern?

**Comment by AshiakerWang** [comment]
Thank you for replying. Appreciate it.

**Comment by jmg-74** [comment]
Related comment of Ilya Mironov about similar implementation in pytorch-dp could also be interesting:
[https://github.com/facebookresearch/pytorch-dp/issues/11](https://github.com/facebookresearch/pytorch-dp/issues/11)

**Comment by AshiakerWang** [comment]
> Related comment of Ilya Mironov about similar implementation in pytorch-dp could also be interesting:
> [facebookresearch/pytorch-dp#11](https://github.com/facebookresearch/pytorch-dp/issues/11)

Thank you very very very much for your kind reply, it helps me lot !
